### ISSUE TITLE
Icewm and VirtualBox resize guest problem

### ISSUE THREAD

**Issue by remfan77** [bbidulock/icewm#401]
I used for years :

- a slackware 14.0
- IceWm 1.3.8
- VirtualBox 4.2.36

All was OK.

Now I'm trying to use an updated linux os, so I installed :
- a gentoo linux distribution
- IceWm 1.5.5
- VirtualBox 4.2.36

I prefer to not update VirtualBox because I have a lot of virtual machines.
All works, but there is a small problem.
When I use **autoresize-guest** feature of VirtualBox, **automatically my guests window begin to be smaller and smaller**.
With IceWm 1.3.8 all is OK also on gentoo.
**This bug has been introduced in 1.3.9**.
With openbox I do not have this problem.
IceWm for me is the best and I would like continue to use it.
Can someone point me what is changed in the resize code ?
Best regards,
Paolo

**Comment by gijsbers** [comment]
We need to know which window properties are involved.
Could you compile this little program as `timing`
with `gcc -O1 -Wall timing.c -o timing`:

```C
#include <stdlib.h>
#include <stdio.h>
#include <sys/time.h>

int main () {
    char buf[BUFSIZ];
    struct timeval start = { 0, 0 };
    gettimeofday(&start, NULL);
    while (fgets(buf, BUFSIZ, stdin)) {
        struct timeval tv = { 0, 0 };
        gettimeofday(&tv, NULL);
        struct timeval dif;
        timersub(&tv, &start, &dif);
        printf("%02ld.%06lu: %s", dif.tv_sec, dif.tv_usec, buf);
    }
    return 0;
}
```

And then run `xprop -spy | ./timing` on the window which is
causing the problems and attach the output?

It is not actually clear if the bug is in IceWM or in VirtualBox.
It could be that VirtualBox 6 is not prone to this problem.
Could you perhaps create a new account, login using that account
and then try to reproduce this using a recent VirtualBox?
If you do please report the actual steps to reproduce it in detail
so we can repeat your exact procedure to set it all up and see it.
That would be needed to start fixing it.

**Comment by remfan77** [comment]
Thanks @gijsbers for your quick answer.
I understand what you mean. For me it is not a very simple install Virtual Box 6 and try to get the log.
I would like not to install other software, but doing so I understand for you is difficult to solve my problem. The version of Virtual Box 4.2.36 is also old, so I understand you do not want to waste your time for old software.
I made a simple video to show you the problem.
https://we.tl/t-wYEPUmELfo
As you can see, I launch ```xprop -spy```, then I enable in Virtual Box 4.2.36 the feature resize guest.
When I enable this feature, the window begins to be smaller and smaller automatically.
During the process of becoming small, there is only a block of log.
When the windows is very very small , there are some blocks of log.
I hope this can help you a little...

**Comment by gijsbers** [comment]
Well, IceWM should support old software, but trying to install an old version of virtualbox and create a new VM takes a lot of work. The video shows that VirtualBox is constantly changing the size and position requests in the WM_NORMAL_HINTS property. That is something it should not do, because IceWM responds to those requests.

We may have a workaround. If you look at the WM_CLASS of that window it shows "Qt-subapplication", "VirtualBox". There is a chance that if you remove the frame that it won't make it smaller. In the IceWM `winoptions` file you could define:

    Qt-subapplication.VirtualBox.dBorder: 0
    Qt-subapplication.VirtualBox.dResize: 0
    Qt-subapplication.VirtualBox.dTitleBar: 0

Restart icewm. You can still resize and move by keys. KeyWinMove, KeyWinSize.

**Comment by remfan77** [comment]
Hi @gijsbers 
Yes, it works. Thanks for support.

**Comment by gijsbers** [comment]
@remfan77 Thanks to your observation that 1.3.9 introduced this behavior, I was able to find the cause and fix it. The implementation of _NET_MOVERESIZE_WINDOW was sketchy and never improved since. Amazing that this survived so long. I expect that you will be able to undo the workaround and see proper behavior if you use this. Thanks a lot for reporting this and digging up when it was introduced.

**Comment by remfan77** [comment]
@gijsbers I cloned repository and I configured and compiled the source code
```
> git branch
icewm-1-4-BRANCH
```
All seems OK with these new changes.
In which versions do you think to introduce this patch ? In all branches 1.4.x, 1.5.x, 1.6.x or only in the last 1.6.x ? Thanks for support

**Comment by gijsbers** [comment]
Happy to hear it now works well. I don't know of reasons to maintain older versions, so 1.6.4 should have this. We already have a handful additional fixes for flaws which were also in 1.5.5.
